### DISCUSSION TITLE
Where to find DDN CLI release info or source code or its CI/build process?

### DISCUSSION THREAD

**Discussion by ev-dev** [hasura/graphql-engine#10604] [Question]
Struggling to find how the `ddn` CLI queries for the latest available version, is there an endpoint or any other easy way to retrieve the published versions per platform/arch? 

The DDN install script uses `latest` in its download URLs and previous versions are accessible too (like https://graphql-engine-cdn.hasura.io/ddn/cli/v4/v2.13.0/cli-ddn-windows-amd64.exe), but it's not clear where to get data on all the published cli assets.

Clearly the `ddn` cli also internally checks/compares/updates to latest version but I can't seem to find the actual source code or even any CI/build info about the DDN cli specifically, can anyone point me in the right direction?

**Comment by hgiasac** [comment]
@ev-dev we update the changelog of all releases on this page.
https://hasura.io/changelog/ddn

The DDN CLI isn't open-source so you can't look at the source code. However, the download URL of most CLI versions has the same pattern:

```
# Linux & Mac
https://graphql-engine-cdn.hasura.io/ddn/cli/v4/<version>/cli-ddn-<os>-<arch>

# windows
https://graphql-engine-cdn.hasura.io/ddn/cli/v4/<version>/cli-ddn-windows-<arch>.exe
```
